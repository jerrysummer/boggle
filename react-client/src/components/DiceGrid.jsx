import React from 'react';
import Die from './Die.jsx';
import { connect } from 'react-redux';

class DiceGrid extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="diceGrid">
        {
          this.props.board.map((die, i) => (<Die
            die={die}
            key={i}
            pos={i}
          />))
        }
      </div>
    );
  }
}

function mapStateToProps({ board }) {
  return { board };
}

export default connect(mapStateToProps)(DiceGrid);
