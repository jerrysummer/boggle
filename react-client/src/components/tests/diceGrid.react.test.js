import React from 'react';
import renderer from 'react-test-renderer';
import DiceGrid from '../DiceGrid';

const board = [{ val: 'A', selected: false },
  { val: 'E', selected: false },
  { val: 'S', selected: false },
  { val: 'Qu', selected: false },
  { val: 'H', selected: false },
  { val: 'N', selected: false },
  { val: 'J', selected: false },
  { val: 'C', selected: false },
  { val: 'T', selected: false },
  { val: 'P', selected: false },
  { val: 'U', selected: false },
  { val: 'U', selected: false },
  { val: 'E', selected: false },
  { val: 'M', selected: false },
  { val: 'N', selected: false },
  { val: 'I', selected: false },
  { val: 'D', selected: false },
  { val: 'N', selected: false },
  { val: 'R', selected: false },
  { val: 'S', selected: false },
  { val: 'D', selected: false },
  { val: 'Y', selected: false },
  { val: 'W', selected: false },
  { val: 'E', selected: false },
  { val: 'Y', selected: false }];

it('renders correctly', () => {
  const component = renderer
    .create(<DiceGrid board={board} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
