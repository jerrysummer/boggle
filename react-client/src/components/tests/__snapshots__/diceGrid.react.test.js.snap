// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="diceGrid"
>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      A
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      E
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      S
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      Qu
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      H
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      N
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      J
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      C
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      T
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      P
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      U
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      U
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      E
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      M
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      N
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      I
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      D
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      N
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      R
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      S
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      D
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      Y
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      W
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      E
    </div>
  </div>
  <div
    className="die"
    onClick={[Function]}
  >
    <div
      className="die-circle"
    >
      Y
    </div>
  </div>
</div>
`;
